File: CineBox/CineBox.Model/ViewModels/Movie.cs
using System;
namespace CineBox.Model.ViewModels
{
    public partial class Movie
    {
        public int Id { get; set; }

        public int GenreId { get; set; }

        public string Title { get; set; } = null!;

        public string Description { get; set; } = null!;

        public DateTime PerformedFrom { get; set; }

        public DateTime PerformedTo { get; set; }

        public string Director { get; set; } = null!;

        public int? PictureId { get; set; }

        public virtual Genre Genre { get; set; } = null!;

        public virtual Picture? Picture { get; set; }
    }
}

File: CineBox/UI/cinebox_mobile/lib/models/Movie/movie.dart
import 'package:cinebox_mobile/models/Actor/actor.dart';
import 'package:cinebox_mobile/models/Picture/picture.dart';
import 'package:json_annotation/json_annotation.dart';

part 'movie.g.dart';

@JsonSerializable()
class Movie {
  int? id;
  int? genreId;
  String? title;
  String? description;
  DateTime? performedFrom;
  DateTime? performedTo;
  int? duration;
  String? director;
  int? pictureId;
  List<Actor>? actors;
  Picture? picture;

  Movie();

  factory Movie.fromJson(Map<String, dynamic> json) => _$MovieFromJson(json);

  Map<String, dynamic> toJson() => _$MovieToJson(this);
}
